### PR TITLE
Remove context.TODO() from client_go function calls

### PR DESCRIPTION
**Pull request by slashpai**
This PR removes context.TODO() from most of the function calls. We still have some more to remove which needs to check further. Since the Go tests have no context wrapping them added a context.Context field to the [Framework](https://github.com/prometheus-operator/prometheus-operator/blob/2b93704fc1d1d5dbc4e7aa98c2a1425a48bbca42/test/framework/framework.go#L52-L61) struct, initialised it to context.Background() and propagated it to tests.

Remaining files which has context.TODO()

```
▒▓    ~/gi/s/prometheus-operator  on   remove_context_todo ⇣1⇡1  find . -name "*go" | xargs grep -l context.TODO       ✔  2.7.17   1.16.4   2.7.3   at admin/openshift-monitoring ⎈  at 18:50:34  ▓▒░
./test/framework/deployment.go
./test/framework/service.go
./test/framework/role-binding.go
./test/framework/cluster_role.go
./test/framework/service_account.go
./test/framework/ingress.go
./test/framework/admission-webhooks.go
./test/framework/cluster_role_binding.go
./test/framework/secret.go
./test/framework/helpers.go
./test/framework/namespace.go
./test/framework/replication-controller.go
./pkg/alertmanager/amcfg_test.go
./pkg/k8sutil/k8sutil_test.go
./pkg/client/informers/externalversions/monitoring/v1/podmonitor.go
./pkg/client/informers/externalversions/monitoring/v1/prometheusrule.go
./pkg/client/informers/externalversions/monitoring/v1/servicemonitor.go
./pkg/client/informers/externalversions/monitoring/v1/alertmanager.go
./pkg/client/informers/externalversions/monitoring/v1/probe.go
./pkg/client/informers/externalversions/monitoring/v1/prometheus.go
./pkg/client/informers/externalversions/monitoring/v1/thanosruler.go
./pkg/client/informers/externalversions/monitoring/v1alpha1/alertmanagerconfig.go

░▒▓    ~/gi/s/prometheus-operator  on   remove_context_todo ⇣1⇡1   
```

Related to https://github.com/prometheus-operator/prometheus-operator/issues/3158

Signed-off-by: Jayapriya Pai <slashpai9@gmail.com>

**Release Note Template (will be copied)**

```release-note:REPLACEME
Removes context.TODO() from most of the files
```

### CODE REVIEW

**Review by dgrisonnet — COMMENTED** [review]
Beware that since you mentioned `Fixes #3158` in the description of this PR, it will close the issue whenever this PR is merged. If you are planning to remove the rest of the context.TODO() occurrences, prefer `Relates to #3158` so that you'll be able to close the issue via a follow-up PR.

**Comment by slashpai** [comment]
> Beware that since you mentioned `Fixes #3158` in the description of this PR, it will close the issue whenever this PR is merged. If you are planning to remove the rest of the context.TODO() occurrences, prefer `Relates to #3158` so that you'll be able to close the issue via a follow-up PR.

Thanks for pointing out that. Is it applicable only for PR description or commit message as well?

**Comment by dgrisonnet** [comment]
It is applicable to both commit messages and descriptions.

**Comment by slashpai** [comment]
Updated commit messages and PR description

File: test/framework/context.go [hunk 1]
```diff
@@ -44,7 +46,8 @@ func (f *Framework) NewTestCtx(t *testing.T) TestCtx {
 
 	id := prefix + "-" + strconv.FormatInt(time.Now().Unix(), 36)
 	return TestCtx{
-		ID: id,
+		ID:  id,
+		Ctx: f.Ctx,
```
[the remark this answers is not in the document]
Comment (slashpai) [reply test/framework/context.go]: I didn't catch that. Thank you this suggestion helped me to remove more context.TODO() :)

**Review by dgrisonnet — APPROVED** [review]
LGTM

**Comment by slashpai** [comment]
@simonpasquier @paulfantom Could you please review? :)

**Review by simonpasquier — APPROVED** [review]
lgtm

**Comment by slashpai** [comment]
Is there a way we can retest only failed ones? e2e prometheus test was passing first attempt. Any known flaky test?

**Comment by simonpasquier** [comment]
I've restarted the tests (you can't do it yourself). #4070 is the flake.

**Comment by simonpasquier** [comment]
thanks!

**Pull request merged**